searchcore/src/vespa/searchcore/proton/attribute/attribute_usage_sampler_functor.cpp

```diff
@@ -2,10 +2,49 @@
 
 #include "attribute_usage_sampler_functor.h"
 #include "attribute_usage_sampler_context.h"
+#include "attribute_config_inspector.h"
 #include <vespa/searchlib/attribute/attributevector.h>
+#include <vespa/searchlib/attribute/loadedenumvalue.h>
+#include <vespa/searchlib/attribute/loadedvalue.h>
+
+using search::attribute::BasicType;
 
 namespace proton {
 
+namespace {
+
+size_t
+get_transient_memory_usage(const search::attribute::Config& old_config,
+                           const search::attribute::Config& current_config,
+                           uint64_t total_value_count)
+{
+    if (current_config.fastSearch()) {
+        if (old_config.fastSearch()) {
+            return sizeof(search::attribute::LoadedEnumAttribute) * total_value_count;
+        } else {
+            switch (old_config.basicType().type()) {
+            case BasicType::Type::INT8:
+                return sizeof(search::attribute::LoadedValue<int8_t>) * total_value_count;
+            case BasicType::Type::INT16:
+                return sizeof(search::attribute::LoadedValue<int16_t>) * total_value_count;
+            case BasicType::Type::INT32:
+                return sizeof(search::attribute::LoadedValue<int32_t>) * total_value_count;
+            case BasicType::Type::INT64:
+                return sizeof(search::attribute::LoadedValue<int64_t>) * total_value_count;
+            case BasicType::Type::FLOAT:
+                return sizeof(search::attribute::LoadedValue<float>) * total_value_count;
+            case BasicType::Type::DOUBLE:
+                return sizeof(search::attribute::LoadedValue<double>) * total_value_count;
+            default:
+                ;
+            }
+        }
+    }
+    return 0u;
+}
+
+}
+
 AttributeUsageSamplerFunctor::AttributeUsageSamplerFunctor(
```

Comment:
Also consider adding unit test for this class.

Reply:
Done.